### PR TITLE
[Java] Format integration tests code

### PR DESCRIPTION
**Pull request by chaokunyang**
-  Format integration tests code
- Upgrade spotless plugin to support JDK 17&21
- Format java code using upgraded spotless plugin

### CODE REVIEW

**Review by theweipeng — APPROVED** [review]

File: java/pom.xml [hunk 1]
```diff
@@ -246,7 +246,8 @@
         <configuration>
           <java>
             <googleJavaFormat>
-              <version>1.7</version>
+              <!-- 1.19.1 support JDK21 -->
+              <version>1.19.1</version>
               <style>GOOGLE</style>
```
Comment (tisonkun) [inline java/pom.xml:251]: I'll suggest we use the plantir style for longer line length and better breaking lines strategy.

Comment (tisonkun) [reply java/pom.xml:251]: You can check this build https://github.com/apache/curator/blob/111f3a6bdcd562db0493e52243e5cec933cc36ef/pom.xml#L933-L954.

Comment (chaokunyang) [reply java/pom.xml:251]: Thanks, I will do this in another PR

Comment (chaokunyang) [reply java/pom.xml:251]: plantir style seems doesn't support 2 spaces: https://github.com/palantir/palantir-java-format/issues/951. I suggest continuing to use google code style

**Comment by nandakumar131** [comment]
Spotless plugin version upgrade breaks `spotless:apply` in Java 8. (#1278)

**Pull request merged**